recipes-bsp/boot0/boot0.bb

```diff
@@ -0,0 +1,39 @@
+DESCRIPTION = "Mainline friendly Secondary Program Loader for Allwinner D1"
+LICENSE = "GPL-2.0-or-later"
+LIC_FILES_CHKSUM = "file://${COREBASE}/meta/files/common-licenses/GPL-2.0-or-later;md5=fed54355545ffd980b814dab4a3b312c"
+
+FILESEXTRAPATHS:prepend := "${THISDIR}/files:"
+SRC_URI = " \
+	git://github.com/smaeul/sun20i_d1_spl;protocol=https;branch=mainline \
+	file://0001-config.mk-provide-path-to-linux-headers.patch \
+	file://0002-riscv-fix-build-with-binutils-2.38.patch \
+"
+
+PV = "1.0-dev"
+SRCREV = "0ad88bfdb723b1ac74cca96122918f885a4781ac"
+
+S = "${WORKDIR}/git"
+
+CFLAGS += " --sysroot=${RECIPE_SYSROOT}"
+
+EXTRA_OEMAKE = "CROSS_COMPILE=${TARGET_PREFIX} p=sun20iw1p1 mmc"
+EXTRA_OEMAKE += "STAGING_INCDIR=${STAGING_INCDIR_NATIVE}"
+
+GCC_INCLUDE = "${RECIPE_SYSROOT_NATIVE}${libdir}/${TARGET_SYS}/gcc/${TARGET_SYS}/*/include"
+
+inherit deploy
+
+do_configure () {
+	cp -r ${GCC_INCLUDE}/* ${STAGING_INCDIR_NATIVE}
```

Comment:
this either.

Reply:
This is necessary to avoid compilation errors. The `STAGING_INCDIR_NATIVE` doesn't have a `stdbool.h` file and others which I don't remember now:
```
| In file included from fdt_wip.c:6:
| libfdt_env.h:10:10: fatal error: stdbool.h: No such file or directory
|    10 | #include <stdbool.h>
|       |          ^~~~~~~~~~~
| compilation terminated.
```

This is my workaround for this problem.

Reply:
Please apply patches from these pulls https://github.com/smaeul/sun20i_d1_spl/pull/8 and https://github.com/smaeul/sun20i_d1_spl/pull/7, after these patches your recipe should simplify dramatically, and all you would need perhaps is edit LDFLAGS  something like `EXTRA_OEMAKE += 'LDFLAGS='` would do it.